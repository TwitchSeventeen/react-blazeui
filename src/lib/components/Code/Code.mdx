---
name: Code
route: /typography/code
menu: Typography
---

import { Playground, PropsTable } from 'docz'
import Code from './index';

# Code

Used to display a snippet of code.

## Basic usage
By default the code snippet is displayed inline.
<Playground>
    <Code>
        lorem ipsum dolor
    </Code>
</Playground>

## Multiline
To display a block-level code snippet, use the **multiline** prop.
<Playground>
    <Code multiline >
        hello <br />
        world
    </Code>
</Playground>

## Properties
<PropsTable of={Code} />