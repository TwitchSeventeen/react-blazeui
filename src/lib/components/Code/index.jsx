import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Code (props) {
  const { children, multiline } = props;
  const classes = classNames('u-code', { 'u-code--multiline': multiline });
  return (
      <code className={classes}>
        { children }
      </code>
  );
}

Code.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Defines a multi-line code snippet
   */
  multiline: PropTypes.bool,
};

Code.defaultProps = {
  children: null,
  multiline: false,
};

export default Code;