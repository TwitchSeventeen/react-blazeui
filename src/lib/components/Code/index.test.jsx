import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Code from './index';

describe('Code', () => {
  it('should be defined', () => {
    expect(Code).toBeDefined();
  });

  it('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
        <Code>
          lorem ipsum dolor
        </Code>
    );

    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly a multi-line code snippet', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
        <Code multiline>
          lorem ipsum dolor
          sit amet
        </Code>
    );

    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
