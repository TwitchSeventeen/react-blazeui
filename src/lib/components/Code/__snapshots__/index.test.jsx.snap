// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code renders correctly 1`] = `
<code
  className="u-code"
>
  lorem ipsum dolor
</code>
`;

exports[`Code renders correctly a multi-line code snippet 1`] = `
<code
  className="u-code u-code--multiline"
>
  lorem ipsum dolor sit amet
</code>
`;
